src/server/pfs/server/commit_store.go

```diff
@@ -243,12 +248,34 @@ func setDiff(tx *pachsql.Tx, tr track.Tracker, commit *pfs.Commit, id fileset.ID
 	if err := tr.CreateTx(tx, oid, pointsTo, track.NoTTL); err != nil {
 		return errors.EnsureStack(err)
 	}
+	if exists, err := getRepoExists(tx, commit); err != nil {
```

Comment:
Can we be more specific and check that this commit exists?

Reply:
We can, but after testing I don't think we should. I think there are situations around making commits(might be around makeEmptyCommit) where checking the commit prevents eventual consistency because the commit is being created at roughly the same time as the commit_total is being set. I checked the commit_totals table and get missing rows when checking the commit, and the tests fail after checking for specifically commits. 
I think a creation race like that with the whole repo is way less likely, but let me know if you what you think.